src/ORM/DataList.php

```diff
@@ -1179,33 +1185,53 @@ private function fetchEagerLoadHasOne(
                 }
             }
             if (!$added) {
-                throw new LogicException("Couldn't find parent for record $fetchedID on has_one relation $relationName");
+                throw new LogicException("Couldn't find parent for record $fetchedID on $relationType relation $relationName");
             }
         }
 
+        // NOTE: Unlike the other relation types, we don't have to explicitly fill empty DataObject records
+        // into the has_one components - DataObject does that for us in getComponent() without any extra
+        // db calls.
+
         return [$fetchedRecords, $fetchedIDs];
     }
 
     private function fetchEagerLoadBelongsTo(
         Query|array $parents,
         array $parentIDs,
-        string $belongsToIDField,
+        array $component,
         string $relationDataClass,
         string $relationChain,
-        string $relationName
+        string $relationName,
+        string $relationType
     ): array {
+        $belongsToIDField = $component['joinField'];
         // Get ALL of the items for this relation up front, for ALL of the parents
         // Fetched as an array to avoid sporadic additional queries when the DataList is looped directly
         $fetchedRecords = DataObject::get($relationDataClass)->filter([$belongsToIDField => $parentIDs])->toArray();
         $fetchedIDs = [];
-
+        $foundParents = [];
 
         // Add fetched record to the correct place
         foreach ($fetchedRecords as $fetched) {
             $fetchedIDs[] = $fetched->ID;
             $parentID = $fetched->$belongsToIDField;
-            $this->addEagerLoadedDataToParent($parents, $parentID, $relationChain, $relationName, $fetched, 'has_one');
-        }
+            $foundParents[] = $parentID;
+            $this->addEagerLoadedDataToParent($parents, $parentID, $relationChain, $relationName, $fetched, $relationType);
+        }
+
+        // Load empty DataObject records into any parents which have no child records
+        $emptyRelations = array_diff($parentIDs, $foundParents);
```

Comment:
```suggestion
        $missingParentIDs = array_diff($parentIDs, $foundParentIDs);
```

Reply:
Both `emptyRelations` and `missingParentIDs` are correct.... but not worth discussing, I've made the change.